plugins/woocommerce/src/Blocks/BlockTypes/MiniCart.php

```diff
@@ -282,7 +314,92 @@ public function print_lazy_load_scripts() {
 			);
 		}
 
-		$data                          = rawurlencode( wp_json_encode( $this->scripts_to_lazy_load ) );
+		$this->add_inline_script_data();
+	}
+
+	/**
+	 * Load the frontend dependencies cache when the cart has items.
+	 *
+	 * @return string|null String of JSON data.
+	 */
+	protected function load_frontend_dependencies_cache() {
+		if ( wp_is_development_mode( 'plugin' ) ) {
+			return null;
+		}
+
+		$cache = get_site_transient(
+			$this->is_cart_empty
+				? self::EMPTY_CART_FRONTEND_DEPENDENCIES_CACHE_KEY
+				: self::NON_EMPTY_CART_FRONTEND_DEPENDENCIES_CACHE_KEY
+		);
+
+		$current_version = array(
+			'woocommerce' => WOOCOMMERCE_VERSION,
+			'wordpress'   => get_bloginfo( 'version' ),
+		);
+
+		if ( isset( $cache['version'] ) && $cache['version'] === $current_version ) {
```

Comment:
Do you think we should add a check to verify `$cache['data']` exists and it's a string? Maybe I'm being unnecessarily cautious, but given that it's a value that comes from the database, users or other plugins might have messed up with its format.

Reply:
I think that's a reasonable precaution to take.  Added some checks in https://github.com/woocommerce/woocommerce/pull/52001/commits/19276cb0ec99be82f98327b3e9fdef7af47b724a.